Katana/Store/ActionLinker.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  ActionLinker.swift
+//  Katana
+//
+//  Created by Riccardo Cipolleschi on 11/01/2017.
+//  Copyright © 2017 Bending Spoons. All rights reserved.
+//
+
+import Foundation
+
+/**
+ The engine that dispatches all the actions that depends on a triggered source action.
+ This is invoked every time an action is dispatched. 
+ The ActionLinker is responsible to check if there are actions that must be issued after a source actionis dispatched.
+ For each of this actions, it checks if the conditions are met: if yes, they are dispatched.
+ */
+struct ActionLinker {
+
+  /**
+   Dictionary used to store the links.
+   We use a dictionary to make the access efficient. 
+   The key is a String because Action does not implement Hashable.
+   All the conversion between Action.Type and String is handled internally, 
+   to leverage the type checking system.
+   */
+  let links: [String : [LinkeableAction.Type]]
+  
+  /**
+   Initializer.
+   The first element of LinkedActions is an Action.Type and not a String to enforce type safety and type checking. 
+   The string conversion is entirely handled by the ActionLinker.
+   
+   - parameter links: the array of ActionLinks to register.
+   */
+  init(links: [ActionLinks]) {
+    var tmpLinks: [String: [LinkeableAction.Type]] = [:]
+    
+    /// Create the dictionary for a fast and efficient access to the actions.
+    for tuple in links {
+      let actionType = String(reflecting: tuple.source)
```

Comment:
It is not about saving lines of code. It is about avoid to update it somewhere and forget to update it in another place